spec/units/polling_job_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+describe ActiveEncode::PollingJob do
+  include ActiveJob::TestHelper
+
+  before do
+    class PollingEncode < ActiveEncode::Base
+      include ActiveEncode::Polling
+      after_status_update ->(encode) { encode.history << "PollingEncode ran after_status_update" }
+      after_error ->(encode) { encode.history << "PollingEncode ran after_error" }
+      after_cancelled ->(encode) { encode.history << "PollingEncode ran after_cancelled" }
+      after_complete ->(encode) { encode.history << "PollingEncode ran after_complete" }
+
+      def history
+        @history ||= []
+      end
+    end
+  end
+
+  after do
+    Object.send(:remove_const, :PollingEncode)
+  end
+
+  describe '#perform' do
+    let(:encode) { PollingEncode.create("sample.mp4").tap { |encode| encode.state = state } }
+    let(:poll) { ActiveEncode::PollingJob.new }
+    subject { encode.history }
+
```

Comment:
so tap allows one to set instance variables with let, good to know.

Reply:
`tap` allows you to operate on a method's return value while still returning it from the tap.
It's useful here since `encode.state` doesn't return the `PollingEncode` so tap avoids doing:
```
let(:encode) do
  encode = PollingEncode.create("sample.mp4")
  encode.state = state
  encode
end
```
http://seejohncode.com/2012/01/02/ruby-tap-that/